added reference page for  block (#35112)
hashicorp/terraform: website/docs/language/moved.mdx

---
page_title: moved block configuration reference

> description: Learn about the `moved` block that you can specify in Terraform configurations. The `moved` block programmatically changes the location of resource.

Reviewer: ```suggestion
description: Learn about the `moved` block that you can specify in Terraform configurations. The `moved` block programmatically changes the location of a resource.
```

---

# Moved block configuration reference

This topic provides reference information for the `moved` block. 

## Introduction

The `moved` block programmatically changes the address of a resource. Refer to [Refactoring](/terraform/language/modules/develop/refactoring) for details about how to use the `moved` block in your Terraform configurations.

## Configuration model

The following list outlines field hierarchy, language-specific data types, and requirements in the `moved` block. 

- [`moved`](#moved): map
  - [`from`](#moved): string
  - [`to`](#moved): string

## Complete configuration

When every field is defined, a `moved` block has the following form:

```hcl

> 

Reviewer: I think we can remove the extra line here?
```suggestion
```

moved = {
    from = <old address for the resource>
    to = <new address for the resource>
}
```

## Specification

This section provides details about the fields you can configure in the `moved` block.

### `moved`

Map that specifies addresses for the resource. The following table describes the fields you can set in the `moved` block.

| Field | Description | Type | Required |
| ---   | ---         | ---  | ---      |

> | `from` | Specifies the previous address of the resource addresses using a syntax that allows Terraform to select modules, resources, and resources inside child modules. | string | required |

Reviewer: ```suggestion
| `from` | Specifies a resource's previous address, using a syntax that allows Terraform to select modules, resources, and resources inside child modules. | string | required |
```

> | `to` | Sepcifies the current address of the resource using a syntax that allows Terraform to select modules, resources, and resources inside child modules. | string | required |

Reviewer: ```suggestion
| `to` | Specifies the new address to relocate the resource to, using a syntax that allows Terraform to select modules, resources, and resources inside child modules. | string | required |
```

> Before creating a new plan for the resource specified in the `to` field, Terraform checks the state for an existing object at the address specified in the `from` field. Terraform renames existing objects to the string specified in the `to` field then creates a plan. The plan directs Terraform to provision the resource specified in the `from` field as the resource specified in the `to` field. As a result, Terraform does not destroy the resource during the Terraform run.

Reviewer: ```suggestion
Before creating a new plan for the resource specified in the `to` field, Terraform checks the state for an existing object at the address specified in the `from` field. Terraform renames existing objects to the string specified in the `to` field and then creates a plan. The plan directs Terraform to provision the resource specified in the `from` field as the resource specified in the `to` field. As a result, Terraform does not destroy the resource during the Terraform run.
```

## Example

> The following example moves and AWS instance from address `aws_instance.a` to `aws_instance.b`:

Reviewer: ```suggestion
The following example moves an AWS instance from address `aws_instance.a` to `aws_instance.b`:
```
